### PR TITLE
Add general filter that Devs can use to skip some actions from syncing

### PR DESCRIPTION
**Pull request by enejb**
#### Changes proposed in this Pull Request:
- Adds a filter that devs can use to stop an item from being send over to .com. 
-------------------
TODO
- [ ] create a way for devs to not sync posts and comments during full sync. 

-------------------
- [x] Make sure your changes respect [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- [x] Did you create a **new action or filter**? [Add inline documentation](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#4-hooks-actions-and-filters) to help others understand how to use the action or the filter.
- [x] Create **[unit tests](https://github.com/Automattic/jetpack/tree/master/tests)** if you can. If you're not familiar with Unit Testing, you can check [this tutorial](https://pippinsplugins.com/series/unit-tests-wordpress-plugins/).

### CODE REVIEW

File: sync/class.jetpack-sync-queue.php [hunk 1]
```diff
@@ -43,6 +43,21 @@ function __construct( $id ) {
...
+		 * Filters whether to skip adding an item to the sync que.
+		 *
+		 * Passing true to the filter will prevent the item to not be added to the que.
+		 * $item = array( $action, $args, $user_id, $microtime );
+		 *
+		 * @since 4.1.0
+		 *
+		 * @see Jetpack_Sync_Queue->add()
+		 */
+		if ( apply_filters( 'jetpack_skip_sync_item', false, $item ) ) {
```
Comment (jeherve) [inline sync/class.jetpack-sync-queue.php]: Could you add the 2 parameters to the docblock?

File: sync/class.jetpack-sync-queue.php [hunk 2]
```diff
@@ -66,14 +81,21 @@ function add_all( $items ) {
 		$rows = array();
 
 		for ( $i = 0; $i < count( $items ); $i += 1 ) {
+			if ( apply_filters( 'jetpack_skip_sync_item', false, $items[ $i ]) ) {
```
Comment (jeherve) [inline sync/class.jetpack-sync-queue.php]: Could you add a small docblock here, like `/** This action is already documented in sync/class.jetpack-sync-queue.php */`?

File: sync/class.jetpack-sync-client.php [hunk 3]
```diff
@@ -300,7 +300,7 @@ function action_handler() {
 		) {
 			return;
 		}
-
+
```
Comment (jeherve) [inline sync/class.jetpack-sync-client.php]: You'll probably want to remove this whitespace.

**Comment by enejb** [comment]
Thanks for the review @jeherve.

**Comment by enejb** [comment]
Working on an improved version of this pr as we talked about in the meetup. Tracking a blacklist is not so straight forward as initially thought.

**Pull request closed without merging**